### DISCUSSION TITLE
b站如何设置点击新链接在当前页面打开，而不是打开一个新的页面？

### DISCUSSION THREAD

**Discussion by billowssun** [the1812/Bilibili-Evolved#3944] [提问 / Q&A]
因为我想把b站安装成一个应用，在一个窗口跳转就好了，但是每次点一个链接都会打开新窗口很不方便。

**Comment by Ethkuil** [comment]
理论基础：[HTML \<a\> target Attribute](https://www.w3schools.com/tags/att_a_target.asp)

然后把页面上的每个<a>元素都检查、修改一遍即可。

**Comment by Geeyun-JY3** [comment]
相关需求应该有通用油猴脚本。

不过既然你是想将 b 站网页版安装成应用，可以看看 #736

**Comment by the1812** [comment]
猜猜 b 站有多少个用 js 开链接的按钮（

**Comment by Youngcius** [comment]
https://greasyfork.org/en/scripts/389127-bilibili%E8%B6%85%E9%93%BE%E6%8E%A5%E4%B8%8D%E6%96%B0%E5%BC%80%E6%A0%87%E7%AD%BE%E9%A1%B5%E7%9B%B4%E6%8E%A5%E5%9C%A8%E5%BD%93%E5%89%8D%E9%A1%B5%E9%9D%A2%E6%89%93%E5%BC%80

这个脚本

**Comment by billowssun** [comment]
太感谢了~
________________________________
发件人: Youngcius ***@***.***>
发送时间: 2024年1月21日 6:07
收件人: the1812/Bilibili-Evolved ***@***.***>
抄送: billows ***@***.***>; State change ***@***.***>
主题: Re: [the1812/Bilibili-Evolved] b站如何设置点击新链接在当前页面打开，而不是打开一个新的页面？ (Discussion #3944)


https://greasyfork.org/en/scripts/389127-bilibili%E8%B6%85%E9%93%BE%E6%8E%A5%E4%B8%8D%E6%96%B0%E5%BC%80%E6%A0%87%E7%AD%BE%E9%A1%B5%E7%9B%B4%E6%8E%A5%E5%9C%A8%E5%BD%93%E5%89%8D%E9%A1%B5%E9%9D%A2%E6%89%93%E5%BC%80

这个脚本

―

**Comment by Jingyuan-Zheng** [comment]
我自己写了一个脚本，我觉得很完美哈哈哈。见：https://github.com/ChingyuanCheng/bilibili-no-new-tab
